data-serving/scripts/export-data/transform.py

```diff
@@ -220,13 +224,88 @@ def convert_row(row: dict[str, Any]) -> Optional[dict[str, Any]]:
     return row
 
 
+class Writer:
+    "Generic writer class for different formats"
+
+    VALID_FORMATS = ["csv", "tsv", "json", "jsonl"]
+
+    def __init__(self, fmt, output, fields):
+        self.fmt = fmt
+        if fmt not in self.VALID_FORMATS:
```

Comment:
Can use [choices parameter](https://docs.python.org/3/library/argparse.html#choices) instead.
We also might want to consider a better argument parsing library like [click](https://click.palletsprojects.com/en/8.0.x/) at some point.